### PR TITLE
test: Fix intermittent issue in p2p_feefilter

### PR DESCRIPTION
**Pull request by maflcko**
Fixes:

```
Traceback (most recent call last):
  File "test/functional/test_framework/test_framework.py", line 126, in main
    self.run_test()
  File "test/functional/p2p_feefilter.py", line 63, in run_test
    self.test_feefilter()
  File "test/functional/p2p_feefilter.py", line 117, in test_feefilter
    txids = [miniwallet.send_self_transfer(fee_rate=Decimal('0.00020000'), from_node=node1)['wtxid'] for _ in range(3)]
  File "test/functional/p2p_feefilter.py", line 117, in <listcomp>
    txids = [miniwallet.send_self_transfer(fee_rate=Decimal('0.00020000'), from_node=node1)['wtxid'] for _ in range(3)]
  File "test/functional/test_framework/wallet.py", line 63, in send_self_transfer
    txid = from_node.sendrawtransaction(tx_hex)
  File "test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "test/functional/test_framework/authproxy.py", line 146, in __call__
    raise JSONRPCException(response['error'], status)
test_framework.authproxy.JSONRPCException: bad-txns-inputs-missingorspent (-25)

### CODE REVIEW

File: test/functional/p2p_feefilter.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def test_feefilter(self):
         txids = [miniwallet.send_self_transfer(fee_rate=Decimal('0.00020000'), from_node=node0)['wtxid'] for _ in range(3)]
         conn.wait_for_invs_to_match(txids)
         conn.clear_invs()
+        self.sync_mempools()  # must be sure node 1 has received all txs
```
Comment (fjahr) [inline test/functional/p2p_feefilter.py:114]: I am not sure if we still need the same on L102 then?

Also the comment above seems incorrect: There are three transactions being sent, not one.

**Review by fjahr — COMMENTED** [review]
utACK fa3af2c0d3b3ed185fb8c6105dc2d26dc66632af

This certainly helps but I had some questions.

File: test/functional/p2p_feefilter.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/p2p_feefilter.py:114]: If miniwallet is spending the change output of the previous call (which has been sent on node0), the same tx needs to be sent to node1 for the change output to be spendable there as well.

Generally, we need this everywhere where `from_node` changes between invocations.

Comment (MarcoFalke) [reply test/functional/p2p_feefilter.py:114]: > Also the comment above seems incorrect: There are three transactions being sent, not one.

Thanks, fixed typo

**Comment by guggero** [comment]
ACK fa5a91a3

**Pull request merged**